crates/engine_bibtex/src/c_api/exec.rs

```diff
@@ -779,35 +649,1835 @@ fn rs_check_command_execution(ctx: &mut ExecCtx) -> Result<(), BibtexError> {
+
+                let field = globals.other.field(field_ptr);
+                if field == 0 {
+                    ctx.push_stack(ExecVal::Missing(globals.hash.text(ex_fn_loc)));
+                } else {
+                    ctx.push_stack(ExecVal::String(field));
+                }
+                Ok(())
+            }
+        }
+        FnClass::IntEntryVar => {
+            if !ctx.mess_with_entries {
+                bst_cant_mess_with_entries_print(ctx, globals.pool)
+            } else {
+                ctx.push_stack(ExecVal::Integer(globals.entries.ints(
+                    globals.cites.ptr() * globals.entries.num_ent_ints()
+                        + globals.hash.ilk_info(ex_fn_loc) as usize,
+                )));
+                Ok(())
+            }
+        }
+        FnClass::StrEntryVar => {
+            if !ctx.mess_with_entries {
+                bst_cant_mess_with_entries_print(ctx, globals.pool)
+            } else {
+                let str_ent_ptr = globals.cites.ptr() * globals.entries.num_ent_strs()
+                    + globals.hash.ilk_info(ex_fn_loc) as usize;
+                let str = globals.entries.strs(str_ent_ptr);
+                ctx.push_stack(ExecVal::String(globals.pool.add_string_raw(str)?));
+                Ok(())
+            }
+        }
+        FnClass::IntGlblVar => {
+            ctx.push_stack(ExecVal::Integer(globals.hash.ilk_info(ex_fn_loc)));
+            Ok(())
+        }
+        FnClass::StrGlblVar => {
+            let str_glb_ptr = globals.hash.ilk_info(ex_fn_loc) as usize;
+            let str_ptr = globals.globals.str_ptr(str_glb_ptr);
+            if str_ptr > 0 {
+                ctx.push_stack(ExecVal::String(str_ptr));
+            } else {
+                let str = globals.globals.str(str_glb_ptr);
+                ctx.push_stack(ExecVal::String(globals.pool.add_string_raw(str)?));
+            }
+            Ok(())
+        }
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn execute_fn(ctx: *mut ExecCtx, ex_fn_loc: HashPointer) -> CResult {
+    with_buffers_mut(|buffers| {
+        with_pool_mut(|pool| {
+            with_hash_mut(|hash| {
+                with_entries_mut(|entries| {
+                    with_globals_mut(|globals| {
+                        with_bibs_mut(|bibs| {
+                            with_cites(|cites| {
+                                with_other(|other| {
+                                    let mut globals = GlobalItems {
+                                        buffers,
+                                        pool,
+                                        hash,
+                                        entries,
+                                        globals,
+                                        bibs,
+                                        cites,
+                                        other,
+                                    };
+                                    rs_execute_fn(&mut *ctx, &mut globals, ex_fn_loc)
```

Comment:
Amazing.

Reply:
My next MR makes this look a little nicer - though really, soon all the `with_*` methods will go away entirely.